recipes/mosquitto/2.x/conanfile.py

```diff
@@ -0,0 +1,177 @@
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("edl-v10", src=self._source_subfolder, dst="licenses")
+        self.copy("epl-v20", src=self._source_subfolder, dst="licenses")
+        self.copy("LICENSE.txt", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "res"), "*.example")
+        if not self.options.shared:
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.so*")
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.dylib")
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "*.dll")
+        elif self.options.shared and self.settings.compiler == "Visual Studio":
+            self.copy("mosquitto.lib", src=os.path.join(self._build_subfolder, "lib"), dst="lib")
+            if self.options.build_cpp:
+                self.copy("mosquittopp.lib", src=os.path.join(self._build_subfolder, "lib"), dst="lib")
+
+    def package_info(self):
+        libsuffix = "" if self.options.shared else "_static"
+        self.cpp_info.components["libmosquitto"].names["pkg_config"] = "libmosquitto"
+        self.cpp_info.components["libmosquitto"].libs = ["mosquitto" + libsuffix]
+        if self.options.ssl:
+            self.cpp_info.components["libmosquitto"].requires = ["openssl::openssl"]
+        if self.settings.os == "Linux":
+            self.cpp_info.components["libmosquitto"].system_libs = ["pthread", "m"]
+        elif self.settings.os == "Windows":
+            self.cpp_info.components["libmosquitto"].system_libs = ["ws2_32"]
+
+        if self.options.build_cpp:
+            self.cpp_info.components["libmosquittopp"].names["pkg_config"] = "libmosquittopp"
+            self.cpp_info.components["libmosquittopp"].libs = ["mosquittopp" + libsuffix]
+            self.cpp_info.components["libmosquittopp"].requires = ["libmosquitto"]
+            if self.settings.os == "Linux":
+                self.cpp_info.components["libmosquittopp"].system_libs = ["pthread", "m"]
+            elif self.settings.os == "Windows":
+                self.cpp_info.components["libmosquittopp"].system_libs = ["ws2_32"]
+
+        if self.options.broker:
+            self.cpp_info.components["broker"].libdirs = []
+            self.cpp_info.components["broker"].include_dirs = []
+            bin_path = os.path.join(self.package_folder, "bin")
+            self.output.info("Appending PATH env var with : {}".format(bin_path))
+            self.env_info.PATH.append(bin_path)
+            if self.options.websockets:
+                self.cpp_info.components["broker"].requires.append("libwebsockets::libwebsockets")
+            if self.settings.os == "Linux":
```

Comment:
@ericLemanissier will thank you
```suggestion
            if self.settings.os in ("FreeBSD", "Linux"):
```